### ISSUE TITLE
[Bug] Cannot set headers after they are sent to the client

### ISSUE THREAD

**Issue by qlaffont** [nguyenbatranvan/vite-multiple-assets#7]
Hi,
I am using this plugin and I have a log who is emitted on vite. This issue is not fatal, I can access to my assets but i prefer to share it to fix it : 

```bash
GET http://localhost:3100/fonts/Steradian/steradian-rg-webfont.woff2
An unhandled error occured: Error: Cannot set headers after they are sent to the client
    at new NodeError (node:internal/errors:405:5)
    at ServerResponse.setHeader (node:_http_outgoing:648:11)
    at handleNodeResponse (file:///home/quentin/Documents/dev/madeformed/starter-web/node_modules/.pnpm/solid-start@0.3.9_@solidjs+meta@0.29.1_@solidjs+router@0.8.3_solid-js@1.8.5_solid-start-node@0.3.9_vite@4.5.0/node_modules/solid-start/node/fetch.js:171:16)
    at startHandler (file:///home/quentin/Documents/dev/madeformed/starter-web/node_modules/.pnpm/solid-start@0.3.9_@solidjs+meta@0.29.1_@solidjs+router@0.8.3_solid-js@1.8.5_solid-start-node@0.3.9_vite@4.5.0/node_modules/solid-start/dev/server.js:140:7)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
```

Regards

**Comment by nguyenbatranvan** [comment]
> Hi, I am using this plugin and I have a log who is emitted on vite. This issue is not fatal, I can access to my assets but i prefer to share it to fix it :
> 
> ```shell
> GET http://localhost:3100/fonts/Steradian/steradian-rg-webfont.woff2
> An unhandled error occured: Error: Cannot set headers after they are sent to the client
>     at new NodeError (node:internal/errors:405:5)
>     at ServerResponse.setHeader (node:_http_outgoing:648:11)
>     at handleNodeResponse (file:///home/quentin/Documents/dev/madeformed/starter-web/node_modules/.pnpm/solid-start@0.3.9_@solidjs+meta@0.29.1_@solidjs+router@0.8.3_solid-js@1.8.5_solid-start-node@0.3.9_vite@4.5.0/node_modules/solid-start/node/fetch.js:171:16)
>     at startHandler (file:///home/quentin/Documents/dev/madeformed/starter-web/node_modules/.pnpm/solid-start@0.3.9_@solidjs+meta@0.29.1_@solidjs+router@0.8.3_solid-js@1.8.5_solid-start-node@0.3.9_vite@4.5.0/node_modules/solid-start/dev/server.js:140:7)
>     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
> ```
> 
> Regards

Can you explain more clearly what the problem is?

**Comment by qlaffont** [comment]
The problem is when i try to access some assets :
The assets is correctfully downloaded BUT an error is display on vite console " Error: Cannot set headers after they are sent to the client"

**Comment by qlaffont** [comment]
any update on this ?

**Comment by nguyenbatranvan** [comment]
i don't know exactly, is the problem with SSR?

**Comment by qlaffont** [comment]
Yes i have ssr enabled

**Comment by nguyenbatranvan** [comment]
> Yes i have ssr enabled

I see you are using solid-start!! Maybe I need time to use my plugin with this library to find the exact cause

**Comment by qlaffont** [comment]
Yes no problem i will create a repro today ! I keep you in touch !

**Comment by qlaffont** [comment]
Here the repo :  https://github.com/qlaffont/solid-start-project

I take an example with favicon.ico

**Comment by nguyenbatranvan** [comment]
> Here the repro : https://github.com/qlaffont/solid-start-project
> 
> I take an example with favicon.ico

tks so much!!

**Comment by nguyenbatranvan** [comment]
> Hi, I am using this plugin and I have a log who is emitted on vite. This issue is not fatal, I can access to my assets but i prefer to share it to fix it :
> 
> ```shell
> GET http://localhost:3100/fonts/Steradian/steradian-rg-webfont.woff2
> An unhandled error occured: Error: Cannot set headers after they are sent to the client
>     at new NodeError (node:internal/errors:405:5)
>     at ServerResponse.setHeader (node:_http_outgoing:648:11)
>     at handleNodeResponse (file:///home/quentin/Documents/dev/madeformed/starter-web/node_modules/.pnpm/solid-start@0.3.9_@solidjs+meta@0.29.1_@solidjs+router@0.8.3_solid-js@1.8.5_solid-start-node@0.3.9_vite@4.5.0/node_modules/solid-start/node/fetch.js:171:16)
>     at startHandler (file:///home/quentin/Documents/dev/madeformed/starter-web/node_modules/.pnpm/solid-start@0.3.9_@solidjs+meta@0.29.1_@solidjs+router@0.8.3_solid-js@1.8.5_solid-start-node@0.3.9_vite@4.5.0/node_modules/solid-start/dev/server.js:140:7)
>     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
> ```
> 
> Regards

Hi!!! I have released version 1.2.7, you can try again!!

**Comment by qlaffont** [comment]
I can confirm it works ! I close the issue !

**Comment by nguyenbatranvan** [comment]
> I can confirm it works ! I close the issue !

i released version `1.2.8`. With ssr i **vite.congfig.ts** 

```
DynamicPublicDirectory(["../../../shared-assets"],{
            ssr:true
        })]
```

**Comment by qlaffont** [comment]
mimeTypes is required when i specify ssr is it normal ? @nguyenbatranvan

**Comment by nguyenbatranvan** [comment]
> mimeTypes is required when i specify ssr is it normal ? @nguyenbatranvan

what!!! mimeTypes not required!!

**Comment by qlaffont** [comment]
![image](https://github.com/nguyenbatranvan/vite-multiple-assets/assets/10044790/30db4d68-66af-4037-8d85-2fd3854991c7)

**Comment by nguyenbatranvan** [comment]
> ![image](https://private-user-images.githubusercontent.com/10044790/290414557-30db4d68-66af-4037-8d85-2fd3854991c7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDI1MzI0NzUsIm5iZiI6MTcwMjUzMjE3NSwicGF0aCI6Ii8xMDA0NDc5MC8yOTA0MTQ1NTctMzBkYjRkNjgtNjZhZi00MDM3LThkODUtMmZkMzg1NDk5MWM3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjE0VDA1MzYxNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTc1ZjkxY2I4MDgzMGQ0NzVlNmViY2ExYTI0ZTZkYTQ3NTcyNWU2MDg1Nzc4Yzc1YWY3ZTNiMWVjYTFlYjRlMTQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.nL890LMmdExtVIHcPWLEXhiZa2maXRCFeowuDRrhMSQ)

i seen!! What is ts version?

**Comment by qlaffont** [comment]
5.2.2

**Comment by nguyenbatranvan** [comment]
> 5.2.2

yess!! Maybe they should all be optional =))

**Comment by nguyenbatranvan** [comment]
> > 5.2.2
> 
> yess!! Maybe they should all be optional =))

fixed v `1.2.9`
